File: Leet-028. Implement strStr()/solution-n^2.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function(haystack, needle) {
    if (haystack === null || needle === null || haystack.length < needle.length) {
        return -1;
    }
    
    if (haystack.length === 0) {
        return needle.length === 0 ? 0 : -1;
    }
    
    for (var i = 0; i < haystack.length - needle.length + 1; i++) {
        for (var j = 0; j < needle.length; j++) {
            if (haystack[i + j] !== needle[j]) {
                break ;
            }
        }
        
        if (j === needle.length) {
            return i;    
        }
    }
    
    return -1;
};

File: Lint-792. Kth Prime Number/solution-n^2.java
public class Solution {
    /**
     * @param n: the number
     * @return: the rank of the number
     */
    public int kthPrime(int n) {
        if (n < 1) {
            return 0;
        }
        
        List<Integer> primes = new ArrayList<>();
        primes.add(2);
        for (int i = 3; i <= n; i++) {
            boolean isPrime = true;
            for (Integer val : primes) {
                if (i % val == 0) {
                    isPrime = false;
                    break ;
                }
            }
            
            if (isPrime) {
                primes.add(i);
            }
        }
        
        return primes.size();
    }
}